samza-api/src/main/java/org/apache/samza/system/SystemFactory.java

```diff
@@ -32,4 +32,16 @@
   SystemProducer getProducer(String systemName, Config config, MetricsRegistry registry);
 
   SystemAdmin getAdmin(String systemName, Config config);
+
+  default SystemConsumer getConsumer(String systemName, Config config, MetricsRegistry registry, String consumerIdPrefix) {
```

Comment:
Should we mark the other methods as @Deprecated to encourage use of the new methods and identify other call sites that need to be updated?

Reply:
Not sure if it is better. With deprecated, user will still have to implement them later. Can it work as we expect?

Reply:
@MabelYC what do you mean? I'm suggesting that we add the annotation to mark them as deprecated (e.g. https://www.baeldung.com/java-deprecated). We will not remove the older methods. This will mean that when some code calls the older methods, the author will get build time warnings that the method they're calling is deprecated, so that they can start using the new (better) method instead.

Reply:
@prateekm  make sense. Marked theme as @deprecated.